modules/apps/static/portal-configuration/portal-configuration-persistence/src/main/java/com/liferay/portal/configuration/persistence/ConfigurationPersistenceManager.java

```diff
@@ -316,6 +333,16 @@ protected void doStore(
 			String pid, @SuppressWarnings("rawtypes") Dictionary dictionary)
 		throws IOException {
 
+		ConfigurationModelListener configurationModelListener = null;
+
+		if ((_serviceTrackerMap != null) && hasPid(pid)) {
```

Comment:
@jorgeferrer @mhan810 I added this check because certain tests were failing with an NPE caused by the next line.  It seems like some classes may run this before it's actually registered as a component?  Is that possible? Is there a better way to do this than what I'm doing on :225?